test/rest/RestAPITest.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package rest;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import common.DBTestUtil;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.Application;
+import play.GlobalSettings;
+import play.libs.WS;
+import play.test.FakeApplication;
+
+import static common.DBTestUtil.*;
+import static common.TestConstants.*;
+import static org.junit.Assert.assertTrue;
+import static play.test.Helpers.fakeApplication;
+import static play.test.Helpers.running;
+import static play.test.Helpers.testServer;
+
+
+/**
+ * <p>
+ * Class aims to exercise all the rest end points exposed by Dr.Elephant
+ * </p>
+ * <p>
+ * A fake application connecting to an in-memory H2 DB is started inside<br>
+ * the test server which runs the test code. The global class is overridden<br>
+ * so that we don't have to go through the regular application start flow.
+ * </p>
+ */
+public class RestAPITest {
+
+  private static final Logger logger = LoggerFactory.getLogger(RestAPITest.class);
+  private static FakeApplication fakeApp;
+
+  @Before
+  public void setup() {
+    Map<String, String> dbConn = new HashMap<String, String>();
+    dbConn.put(DB_DEFAULT_DRIVER_KEY, DB_DEFAULT_DRIVER_VALUE);
+    dbConn.put(DB_DEFAULT_URL_KEY, DB_DEFAULT_URL_VALUE);
+
+    GlobalSettings gs = new GlobalSettings() {
+      @Override
+      public void onStart(Application app) {
+        logger.info("Starting FakeApplication");
+      }
+    };
+
+    fakeApp = fakeApplication(dbConn, gs);
+  }
+
+  /**
+   * <p>
+   * Rest API - Performs search by job ID
+   * <br>
+   * API provides information on the specific job
+   * </p>
+   * <p>
+   * Following assertions are made in the response json
+   *   <ul>Job id</ul>
+   *   <ul>Job name</ul>
+   *   <ul>Job type</ul>
+   * </p>
+   */
+  @Test
+  public void testrestAppResult() {
+    running(testServer(TEST_SERVER_PORT, fakeApp), new Runnable() {
+      public void run() {
+        populateTestData();
```

Comment:
Isn't it enough if we populate the data once in the setup phase?

And clear the data from the db in a @After method.